test/00/godleyResize.sh

```diff
@@ -27,42 +27,92 @@ pass()
 }
 
 trap "fail" 1 2 3 15
+cat >input.py <<EOF
+import sys
+sys.path.insert(0, '$here')
+from pyminsky import minsky
 
-# insert ecolab script code here
-# use \$ in place of $ to refer to variable contents
-# exit 0 to indicate pass, and exit 1 to indicate failure
-cat >input.tcl <<EOF
-proc bgerror x {
-  puts \$x
-  exit 1
-}
+# Function to simulate assertion behavior
+def assert_condition(condition, message="Assertion failed!"):
+    if not condition:
+        raise AssertionError(message)
+
+# Generalized function to find and return the specified item or group type on the canvas
+def findObject(type_name):
+    if type_name == "Group":
+        # Access groups by index, as minsky.model.groups is not directly iterable
+        for i in range(len(minsky.model.groups)):
+            group = minsky.model.groups[0]
+            if group.visible():
+                minsky.canvas.getItemAt(group.x(), group.y())
+                return minsky.canvas.item()
+    else:
+        # Otherwise, search through items
+        for i in range(len(minsky.model.items)):
+            item = minsky.model.items[i]
+            if item.classType() == type_name and item.visible():
+                minsky.canvas.getItemAt(item.x(), item.y())
+                return minsky.canvas.item()
+    return None
+
+# Add a Godley table to the canvas
+minsky.canvas.addGodley()
+
+# Attempt to select the GodleyIcon
+godley_icon = findObject("GodleyIcon")
```

Comment:
I know this is recapitulating the TCL code, but `minsky.canvas.addGodley()` adds the item to `minsky.canvas.itemFocus()`, so there's no need to do findObject here.

I believe `godley_icon=minsky.canvas.itemFocus()` will simply create the reference you need.

There's two different implementations of the python bindings, the older version here in Minsky, and a newer one being developed in EcoLab, and coming to Minsky soon. So the behaviopur may no correspond with what I remember, and needs to be tested.